transport/conn.go

```diff
@@ -343,3 +344,19 @@ func (c *Conn) sendRequest(req frame.Request, compress, tracing bool) (frame.Res
 
 	return resp.Response, resp.Err
 }
+
+func (c *Conn) registerEvents(e []frame.EventType) (responseHandler, error) {
+	h := make(responseHandler)
```

Comment:
I can't grasp the need of doing this, doesn't `sendRequest()` make a responseHandler for you?